### PR TITLE
Ma/client secret null

### PR DESCRIPTION
**Pull request by markusahlstrand**
The client_secret should be null by default so it falls back to the default value

### CODE REVIEW

File: src/routes/management-api/connections.ts [hunk 1]
```diff
@@ -261,12 +261,9 @@ export const connectionRoutes = new OpenAPIHono<{ Bindings: Env }>()
...
       const body = ctx.req.valid("json");
 
-      const connection = await ctx.env.data.connections.create(tenant_id, {
-        ...body,
-        client_secret: body.client_secret || nanoid(),
-      });
+      const connection = await ctx.env.data.connections.create(tenant_id, body);
 
-      return ctx.json(connection);
+      return ctx.json(connection, { status: 201 });
```
Comment (markusahlstrand) [inline src/routes/management-api/connections.ts:266]: Set status code to 201

File: src/adapters/kysely/connections/create.ts [hunk 2]
```diff
@@ -13,10 +13,12 @@ export function create(db: Kysely<Database>) {
       ...params,
       created_at: new Date().toISOString(),
       updated_at: new Date().toISOString(),
-      tenant_id,
     };
 
-    await db.insertInto("connections").values(connection).execute();
+    await db
+      .insertInto("connections")
+      .values({ ...connection, tenant_id })
```
Comment (markusahlstrand) [inline src/adapters/kysely/connections/create.ts:20]: Don't return the tenant_id as it's not really part of the entity

File: src/routes/management-api/connections.ts [hunk 3]
```diff
@@ -261,12 +261,9 @@ export const connectionRoutes = new OpenAPIHono<{ Bindings: Env }>()
       const { "tenant-id": tenant_id } = ctx.req.valid("header");
       const body = ctx.req.valid("json");
 
-      const connection = await ctx.env.data.connections.create(tenant_id, {
-        ...body,
-        client_secret: body.client_secret || nanoid(),
```
Comment (markusahlstrand) [inline src/routes/management-api/connections.ts:266]: If we set the client_id to a nano id the connections won't work..

File: src/adapters/kysely/connections/create.ts (hunk 2 above)
Comment (danielgent) [reply src/adapters/kysely/connections/create.ts:20]: Return? this is writing

File: test/integration/management-api/connections.spec.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+          },
+        },
+      );
+
+    expect(createConnectionResponse.status).toBe(201);
+    const createdConnection = await createConnectionResponse.json();
+
+    const { created_at, updated_at, id, ...rest } = createdConnection;
+
+    expect(rest).toEqual({ name: "apple" });
```
Comment (danielgent) [inline test/integration/management-api/connections.spec.ts:36]: You can and probably should freeze time if you want to check the other fields :+1:

**Review by danielgent — APPROVED** [review]

File: src/adapters/kysely/connections/create.ts (hunk 2 above)
Comment (markusahlstrand) [reply src/adapters/kysely/connections/create.ts:20]: Yepp, but we're returning the entity above. So, by moving it here it's not returned from the function.

File: test/integration/management-api/connections.spec.ts (hunk 4 above)
Comment (markusahlstrand) [reply test/integration/management-api/connections.spec.ts:36]: We should probably validate that they are set a least. Think testing that the time is now might be excessive..

**Pull request merged**